### PR TITLE
Creating a new example

### PR DESCRIPTION
**Pull request by mmalekzadeh**
A new example "dimension_adaptive_neural_architecture"
For human activity recognition using wearable sensor data for dynamic sensor and sampling rate selection.

Current deep neural architectures for processing sensor data are mainly designed for data coming from a fixed set of sensors, with a fixed sampling rate. Changing the dimensions of the input data causes considerable accuracy loss, unnecessary computations, or application failures. To address this problem, here I build a **dimension-adaptive pooling (DAP)** layer that makes deep architectures robust to temporal changes in sampling rate and in sensor availability. DAP operates on convolutional filter maps of variable dimensions and produces an input of fixed dimensions suitable for feedforward and recurrent layers. Building on this architectural improvement, I build a **dimension-adaptive training (DAT)** procedure to generalize over the entire space of feasible data dimensions at the inference time. DAT comprises the random selection of dimensions during the forward passes and optimization with accumulated gradients of several backward passes. I then show how to combine DAP and DAT to transform existing non-adaptive deep architectures into a **Dimension-Adaptive Neural Architecture (DANA)** without altering other architectural aspects. This solution does not need up-sampling or imputation, thus reduces unnecessary computations at inference time. It prevents losses in classification accuracy of one of the state-of-the-art deep architectures, under dynamic sensor availability and varying sampling rates.

Please see this for more info: https://github.com/mmalekzadeh/dana

Thanks
Mohammad Malekzadeh

### CODE REVIEW

**Comment by fchollet** [comment]
Thank you for the PR! This is a very detailed and in-depth script.

Unfortunately, it wouldn't work as a Keras.io code example since its topic is overly specialized and would only be relevant to a niche of our users. Code examples are meant to be didactic demonstrations of common deep learning best practices.

Instead, I would suggest that you distribute the code as a standalone repo on GitHub. You could also consider submitting the layer `DimensionAdaptivePooling` to TF Addons, if you're interested: https://github.com/tensorflow/addons

Let me know if I can be helpful

Best,

Francois

**Comment by mmalekzadeh** [comment]
Many Thanks, @fchollet,  for a detailed response and advice.

* The code is distributed as a standalone tutorial [here](https://colab.research.google.com/github/mmalekzadeh/dana/blob/master/experiment/DANA_a_tutorial.ipynb).
I will also aim to work on the `addons` that seems to be a very useful library for academic people. 

* I see that the `Timeseries` section of the Keras `code examples` is a bit sparse and don't see an example of sensor data, which is pervasive.  In the first section of the published [tutorial](https://colab.research.google.com/github/mmalekzadeh/dana/blob/master/experiment/DANA_a_tutorial.ipynb) I have an example of a common deep learning best practices for sensor data, which is from a highly cited paper in the related literature. If it's of interest, I can pull it as an example, without the previous specialized sections.

Best
Mohammad
